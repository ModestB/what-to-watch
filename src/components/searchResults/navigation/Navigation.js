import React from "react";
import { connect } from "react-redux";

// Action Types
import {
  showPreviousResults,
  getTrendingShows,
  getShowsByGenre,
} from "../../../store/actions/actions";

import classes from "./Navigation.module.scss";

export function Navigation(props) {
  let similarButton = null;
  let navigation = null;

  if (props.displaySinglePage) {
    if (props.singlePageType !== "person") {
      let genreIds = null;
      if (!props.displayedResults[0].genre_ids) {
        genreIds = props.displayedResults[0].genres.map((genre) => {
          return genre.id;
        });
      } else {
        genreIds = props.displayedResults[0].genre_ids;
      }
      similarButton = (
        <button
          className="btn btn--primary btn--small"
          onClick={() => props.getShowsByGenre(genreIds, props.singlePageType)}
        >
          Similar
        </button>
      );
    }
    navigation = (
      <header className={`${classes.navigation}`}>
        <button
          className="btn btn--danger btn--small"
          onClick={() => props.showPreviousResults([...props.searchResults])}
        >
          Go back
        </button>
        <button
          className="btn btn--primary btn--small"
          onClick={() => props.getTrendingShows()}
        >
          Trending
        </button>

        {props.displaySinglePage ? similarButton : null}
      </header>
    );
  }

  return navigation;
}

const mapStateToProps = (state) => {
  return {
    searchResults: state.searchResults,
    displayedResults: state.displayedResults,
    displaySinglePage: state.displaySinglePage,
    singlePageType: state.singlePageType,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    showPreviousResults: (prevResults) =>
      dispatch(showPreviousResults(prevResults)),
    getTrendingShows: () => dispatch(getTrendingShows()),
    getShowsByGenre: (genreIds, singlePageType) =>
      dispatch(getShowsByGenre(genreIds, singlePageType)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Navigation);
