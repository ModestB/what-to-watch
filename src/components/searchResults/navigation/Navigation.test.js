import React from "react";

import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import { Navigation } from "./Navigation";

configure({ adapter: new Adapter() });

describe("<Navigation />", () => {
  let wrapper;
  let displayedResults = [
    {
      id: 181812,
      video: false,
      vote_count: 3460,
      vote_average: 6.5,
      title: "Star Wars: The Rise of Skywalker",
      release_date: "2019-12-18",
      original_language: "en",
      original_title: "Star Wars: The Rise of Skywalker",
      genre_ids: [28, 12, 878],
      backdrop_path: "/jOzrELAzFxtMx2I4uDGHOotdfsS.jpg",
      adult: false,
      overview:
        "The surviving Resistance faces the First Order once again as the journey of Rey, Finn and Poe Dameron continues. With the power and knowledge of generations behind them, the final battle begins.",
      poster_path: "/db32LaOibwEliAmSL2jjDF6oDdj.jpg",
      popularity: 295.608,
      media_type: "movie"
    }
  ];

  beforeEach(() => {
    wrapper = shallow(<Navigation />);
  });

  it("it should match snapshot", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("it should match snapshot if displaySinglePage=true", () => {
    wrapper.setProps({
      displaySinglePage: true,
      displayedResults: displayedResults
    });
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("it should render 2 buttons if displaySinglePage=true", () => {
    wrapper.setProps({
      displaySinglePage: true,
      displayedResults: displayedResults
    });
    expect(wrapper.find(".btn")).toHaveLength(3);
  });

  it("it should match snapshot if displaySinglePage=true && singlePageType = person", () => {
    wrapper.setProps({
      displaySinglePage: true,
      displayedResults: displayedResults,
      singlePageType: "person"
    });
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("it should render 3 buttons if displaySinglePage=true && singlePageType = person", () => {
    wrapper.setProps({
      displaySinglePage: true,
      displayedResults: displayedResults,
      singlePageType: "person"
    });
    expect(wrapper.find(".btn")).toHaveLength(2);
  });
});
