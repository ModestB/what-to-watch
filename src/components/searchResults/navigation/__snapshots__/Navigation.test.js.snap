// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> it should match snapshot 1`] = `""`;

exports[`<Navigation /> it should match snapshot if displaySinglePage=true && singlePageType = person 1`] = `
"<header className=\\"navigation\\">
  <button className=\\"btn btn--danger btn--small\\" onClick={[Function: onClick]}>
    Go back
  </button>
  <button className=\\"btn btn--primary btn--small\\" onClick={[Function: onClick]}>
    Trending
  </button>
</header>"
`;

exports[`<Navigation /> it should match snapshot if displaySinglePage=true 1`] = `
"<header className=\\"navigation\\">
  <button className=\\"btn btn--danger btn--small\\" onClick={[Function: onClick]}>
    Go back
  </button>
  <button className=\\"btn btn--primary btn--small\\" onClick={[Function: onClick]}>
    Trending
  </button>
  <button className=\\"btn btn--primary btn--small\\" onClick={[Function: onClick]}>
    Similar
  </button>
</header>"
`;
